src/renderer/components/badge/badge.tsx

```diff
@@ -3,21 +3,59 @@ import "./badge.scss"
 import React from "react";
 import { cssNames } from "../../utils/cssNames";
 import { TooltipDecoratorProps, withTooltip } from "../tooltip";
+import { observer } from "mobx-react";
+import { autobind } from "../../utils";
+import { observable } from "mobx";
+import { Icon } from "../icon";
 
 interface Props extends React.HTMLAttributes<any>, TooltipDecoratorProps {
   small?: boolean;
   label?: React.ReactNode;
 }
 
 @withTooltip
+@observer
 export class Badge extends React.Component<Props> {
```

Comment:
Please move all new logic to new component based on `Badge` and put to same folder. 
It's simple component which is not intended to display everything, but mostly plain-text labels.

Reply:
I agree that is is designed to display plain-text labels. But I disagree that this logic should be moved out. The ubiquity of `Badge` being used for even things that are most likely large (like annotations) means that this code should be in `Badge`. Plus I do not see any place where `Badge` is used where a user might not want to be able to temporarily see the whole thing.